internal/util/grpcclient/client.go

```diff
@@ -106,61 +112,62 @@ func (c *ClientBase) resetConnection(client interface{}) {
 	c.grpcClient = nil
 }
 
-func (c *ClientBase) connect(ctx context.Context, retryOptions ...retry.Option) error {
-	var kacp = keepalive.ClientParameters{
-		Time:                60 * time.Second, // send pings every 60 seconds if there is no activity
-		Timeout:             6 * time.Second,  // wait 6 second for ping ack before considering the connection dead
-		PermitWithoutStream: true,             // send pings even without active streams
-	}
-
-	var err error
-	var addr string
-	connectServiceFunc := func() error {
-		addr, err = c.getAddrFunc()
-		if err != nil {
-			log.Debug(c.GetRole()+" client getAddr failed", zap.Error(err))
-			return err
-		}
-		opts := trace.GetInterceptorOpts()
-		ctx1, cancel := context.WithTimeout(ctx, 15*time.Second)
-		defer cancel()
-		conn, err2 := grpc.DialContext(ctx1, addr,
-			grpc.WithKeepaliveParams(kacp),
-			grpc.WithInsecure(), grpc.WithBlock(),
-			grpc.WithDefaultCallOptions(
-				grpc.MaxCallRecvMsgSize(c.ClientMaxRecvSize),
-				grpc.MaxCallSendMsgSize(c.ClientMaxSendSize)),
-			grpc.WithUnaryInterceptor(
-				grpcmiddleware.ChainUnaryClient(
-					grpcretry.UnaryClientInterceptor(
-						grpcretry.WithMax(3),
-						grpcretry.WithCodes(codes.Aborted, codes.Unavailable),
-					),
-					grpcopentracing.UnaryClientInterceptor(opts...),
-				)),
-			grpc.WithStreamInterceptor(
-				grpcmiddleware.ChainStreamClient(
-					grpcretry.StreamClientInterceptor(grpcretry.WithMax(3),
-						grpcretry.WithCodes(codes.Aborted, codes.Unavailable),
-					),
-					grpcopentracing.StreamClientInterceptor(opts...),
-				)),
-		)
-		if err2 != nil {
-			return err2
-		}
-		if c.conn != nil {
-			_ = c.conn.Close()
-		}
-		c.conn = conn
-		return nil
-	}
-
-	err = retry.Do(ctx, connectServiceFunc, retryOptions...)
+func (c *ClientBase) connect(ctx context.Context) error {
+	addr, err := c.getAddrFunc()
+	if err != nil {
+		log.Error("failed to get cclient address", zap.Error(err))
+		return err
+	}
+
+	opts := trace.GetInterceptorOpts()
+	dialContext, cancel := context.WithTimeout(ctx, dialTimeout)
+	conn, err := grpc.DialContext(
+		dialContext,
+		addr,
+		grpc.WithInsecure(),
+		grpc.WithBlock(),
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(c.ClientMaxRecvSize),
+			grpc.MaxCallSendMsgSize(c.ClientMaxSendSize)),
+		grpc.WithUnaryInterceptor(
+			grpcmiddleware.ChainUnaryClient(
+				grpcretry.UnaryClientInterceptor(
+					grpcretry.WithMax(3),
+					grpcretry.WithCodes(codes.Aborted, codes.Unavailable),
+				),
+				grpcopentracing.UnaryClientInterceptor(opts...),
+			)),
+		grpc.WithStreamInterceptor(
+			grpcmiddleware.ChainStreamClient(
+				grpcretry.StreamClientInterceptor(grpcretry.WithMax(3),
+					grpcretry.WithCodes(codes.Aborted, codes.Unavailable),
+				),
+				grpcopentracing.StreamClientInterceptor(opts...),
+			)),
+		grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:                keepAliveTime,    // send pings every 60 seconds if there is no activity
```

Comment:
make sure the comment is clear and accurate

Reply:
agreed

Reply:
I will change it in next pr and we can move forward.